### PR TITLE
Feature/2 cloudtrail findings

### PR DESCRIPTION
**Pull request by vifor2**
This PR adds 3 finding rules related to CloudTrails, see #2

- Management events
- Data events
- KMS key

Prorietary portion : https://github.com/nccgroup/ScoutSuite-Proprietary/pull/53

### CODE REVIEW

File: ScoutSuite/providers/aws/services/cloudtrail.py [hunk 1]
```diff
@@ -45,17 +45,23 @@ def parse_trail(self, global_params, region, trail):
...
 
         if trail_details:
             # using trail ARN instead of name as with Organizations the trail would be located in another account
             trail_config['wildcard_data_logging'] = self.data_logging_status(trail_config['TrailARN'],
                                                                              trail_details,
                                                                              api_client)
-
+            
+            for es in api_client.get_event_selectors(TrailName=trail_config['TrailARN'])['EventSelectors']:
+                trail_config['DataEventsEnabled'] = True if len(es['DataResources']) > 0 else False
```
Comment (Aboisier) [inline ScoutSuite/providers/aws/services/cloudtrail.py]: I think we can simplify this line.
```suggestion
                trail_config['DataEventsEnabled'] = len(es['DataResources']) > 0
```

File: ScoutSuite/output/data/html/partials/aws/services.cloudtrail.regions.id.trails.html [hunk 2]
```diff
@@ -26,6 +28,7 @@ <h4>Information</h4>
         </li>
         <li class="list-group-item-text">Log File Validation Enabled: <span id="cloudtrail.regions.{{region}}.trails.{{@key}}.LogFileValidationDisabled">{{LogFileValidationEnabled}}</span></li>
         <li class="list-group-item-text">Destination S3 bucket name: <a href="javascript:showObject('services.s3.buckets.{{bucket_id}}')">{{get_value_at 'services.s3.buckets' bucket_id 'name'}}</a>/{{S3KeyPrefix}}</li>
+        <li class="list-group-item-text">KMS key: <span id="cloudtrail.regions.{{region}}.trails.{{@key}}.KmsKeyId">{{#if KmsKeyId}} true {{else}} false {{/if}}</span></li>
```
Comment (Aboisier) [inline ScoutSuite/output/data/html/partials/aws/services.cloudtrail.regions.id.trails.html]: I believe the `id_suffix` is wrong here too:

```suggestion
        <li class="list-group-item-text">KMS key: <span id="cloudtrail.regions.{{region}}.trails.{{@key}}.cloudtrail-kms-key-unused">{{#if KmsKeyId}} true {{else}} false {{/if}}</span></li>
```

File: ScoutSuite/output/data/html/partials/aws/services.cloudtrail.regions.id.trails.html [hunk 3]
```diff
@@ -10,7 +10,9 @@ <h4>Information</h4>
       <ul>
         <li class="list-group-item-text">Name: {{name}}</li>
         <li class="list-group-item-text">Region: {{region}}</li>
-        <li class="list-group-item-text">Enabled: <span id="cloudtrail.regions.{{region}}.trails.{{@key}}.IsLogging">{{IsLogging}}</span></li>
+        <li class="list-group-item-text">Logging: <span id="cloudtrail.regions.{{region}}.trails.{{@key}}.IsLogging">{{IsLogging}}</span></li>
+        <li class="list-group-item-text">Data events: <span id="cloudtrail.regions.{{region}}.trails.{{@key}}.DataEventsEnabled">{{DataEventsEnabled}}</span></li>
+        <li class="list-group-item-text">Management events: <span id="cloudtrail.regions.{{region}}.trails.{{@key}}.ManagementEventsEnabled">{{ManagementEventsEnabled}}</span></li>
```
Comment (Aboisier) [inline ScoutSuite/output/data/html/partials/aws/services.cloudtrail.regions.id.trails.html]: Last one.

```suggestion
        <li class="list-group-item-text">Management events: <span id="cloudtrail.regions.{{region}}.trails.{{@key}}.cloudtrail-management-events-disabled">{{ManagementEventsEnabled}}</span></li>
```

**Comment by j4v** [comment]
My only comment (and I'm realizing this isn't specific to this PR) is whether value names should be in camel case or underscored (`DataEventsEnabled`, `DataEventsEnabled`, etc.).

**Pull request merged**